### PR TITLE
Revising proxyHoldApplication to remove custom function

### PR DESCRIPTION
**Pull request by stewartbutler**
Revising per comments at https://github.com/istio/istio/pull/28873.

### CODE REVIEW

**Comment by linsun** [comment]
Note: when cherry pick this to 1.8, need to cherry pick #27696 also.

**Comment by linsun** [comment]
This LGTM except lint, @howardjohn could you take a look also?  Thanks!

**Comment by stewartbutler** [comment]
I'm actually not sure how best to resolve this, as the code is there in order to support the deprecated (but not yet removed) API option and the use of the deprecated function is why lint is failing. @howardjohn, do you know how to resolve?

File: operator/pkg/apis/istio/v1alpha1/values_types.proto [hunk 1]
```diff
@@ -898,7 +898,9 @@ message ProxyConfig {
   TypeMapStringInterface lifecycle = 36;
 
   // Controls if sidecar is injected at the front of the container list and blocks the start of the other containers until the proxy is ready
-  google.protobuf.BoolValue holdApplicationUntilProxyStarts = 37;
+  //
+  // Deprecated: replaced by ProxyConfig setting which allows per-pod configuration of this behavior.
```
Comment (howardjohn) [inline operator/pkg/apis/istio/v1alpha1/values_types.proto:902]: can you add it here too https://github.com/istio/istio/blob/d13b044f471871116db3dedd5e4da995b2ca0096/operator/pkg/apis/istio/v1alpha1/validation/validation.go#L99? we don't expose deprecation directly from proto well

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
just add `// nolint: staticcheck` for the lint error

**Pull request merged**